src/econf.py

```diff
@@ -29,426 +30,487 @@ class Econf_err(Enum):
+    elif err == 8:
+        raise Exception(val)
+    elif err == 9:
+        raise SyntaxError(val)
+    elif err == 10:
+        raise SyntaxError(val)
+    elif err == 11:
+        raise SyntaxError(val)
+    elif err == 12:
+        raise SyntaxError(val)
+    elif err == 13:
+        raise ValueError(val)
+    elif err == 14:
+        raise ValueError(val)
+    elif err == 15:
+        raise ValueError(val)
+    elif err == 16:
+        raise PermissionError(val)
+    elif err == 17:
+        raise PermissionError(val)
+    elif err == 18:
+        raise PermissionError(val)
+    elif err == 19:
+        raise PermissionError(val)
+    elif err == 20:
+        raise FileNotFoundError(val)
+    elif err == 21:
+        raise Exception(val)
+    else:
+        raise Exception(val)
+
+
+def _encode_str(string: str | bytes) -> bytes:
+    if isinstance(string, str):
+        string = string.encode("utf-8")
+    elif not isinstance(string, bytes):
+        raise TypeError(f"Input must be a string or bytes")
+    return string
+
+
+def _check_int_overflow(val: int) -> bool:
+    if isinstance(val, int):
+        c_val = c_int64(val)
+        return c_val.value == val
+    else:
+        raise TypeError("parameter is not a nteger")
+
+
+def _check_uint_overflow(val: int) -> bool:
+    if isinstance(val, int) & (val >= 0):
+        c_val = c_uint64(val)
+        return c_val.value == val
+    else:
+        raise TypeError("parameter is not a unsigned integer")
+
+
+def _check_float_overflow(val: float) -> bool:
+    if isinstance(val, float):
+        c_val = c_double(val)
+        return c_val.value == val
+    else:
+        raise TypeError("parameter is not a float")
+
+
 libname = ctypes.util.find_library("econf")
 libeconf = CDLL(libname)
 
 
-def setValue(kf, group, key, value):
+def set_value(kf: c_void_p, group: str, key: str, value: Any) -> Econf_err:
     if isinstance(value, int):
-        res = set_int_value(kf, group, key, value)
-    # elif isinstance(value, long):
-    #     res = econf_setInt64Value(kf, group, key, value)
-    # elif isinstance(value, uint):
-    #     res = econf_setUIntValue(kf, group, key, value)
-    # elif isinstance(value, ulong):
-    #     res = econf_setUInt64Value(kf, group, key, value)
+        res = set_int64_value(kf, group, key, value)
     elif isinstance(value, float):
-        res = set_float_value(kf, group, key, value)
-    # elif isinstance(value, double):
-    #     res = econf_setDoubleValue(kf, group, key, value
+        res = set_double_value(kf, group, key, value)
     elif isinstance(value, str):
         res = set_string_value(kf, group, key, value)
     elif isinstance(value, bool):
         res = set_bool_value(kf, group, key, value)
+    else:
+        raise TypeError(f"\"value\" parameter is not one of the supported types")
     return res
 
 
-def read_file(file_name: bytes, delim: bytes, comment: bytes):
+def read_file(file_name: str, delim: str, comment: str) -> c_void_p:
```

Comment:
eventually we should not expose the c_void_p, and create some class like in the documentation